lib/features/colors/color_screen.dart

```diff
@@ -5,105 +5,43 @@ import 'package:color_aap/features/colors/color_logic.dart';
-  Color backgroundColor = Colors.black;
-  Color appBarColor = Colors.white;
-  Color textColor = Colors.white;
-  int counter = 0;
-  late final ColorLogic _colorLogic =
-      serviceLocator.get<ColorLogic>(param1: widget.email);
-
-  @override
-  void initState() {
-    super.initState();
-
-    _loadAndShowUserColors();
-  }
-
-  /// Loads user's saved color preferences from local storage
-  Future<void> _loadAndShowUserColors() async {
-    final userColors = await _colorLogic.loadUserColors();
-    if (userColors != null) {
-      setState(() {
-        backgroundColor = userColors.backgroundColor;
-        appBarColor = userColors.appBarColor;
-        textColor = userColors.textColor;
-      });
-    }
-  }
-
-  /// Changes the background color and saves the change
-  void _changeColor(Color color) {
-    setState(() => backgroundColor = color);
-    _colorLogic.saveColor(
-      backgroundColor: backgroundColor,
-      appBarColor: appBarColor,
-      textColor: textColor,
-    );
-  }
-
-  /// Changes the app bar color and saves the change
-  void _changeColorAppBar(Color color) {
-    setState(() => appBarColor = color);
-    _colorLogic.saveColor(
-      backgroundColor: backgroundColor,
-      appBarColor: appBarColor,
-      textColor: textColor,
-    );
-  }
-
-  /// Changes the text color and saves the change
-  void _changeColorText(Color color) {
-    setState(() => textColor = color);
-    _colorLogic.saveColor(
-      backgroundColor: backgroundColor,
-      appBarColor: appBarColor,
-      textColor: textColor,
-    );
-  }
+ const ColorScreen({required this.email, super.key});
 
   @override
   Widget build(BuildContext context) {
-    return GestureDetector(
-      onTap: () {
-        _changeColor(_colorLogic.generateRandomColor());
-        setState(() => counter++);
-        _changeColorText(_colorLogic.generateRandomColor());
-        _changeColorText(_colorLogic.generateRandomColor());
-      },
-      child: Scaffold(
-        appBar: _ColorAppBar(
-          appBarColor: appBarColor,
-          email: widget.email,
-          onTap: () {
-            setState(() => counter = 0);
-            _changeColorAppBar(_colorLogic.generateRandomColor());
-          },
-        ),
-        body: _ColorBody(
-          backgroundColor: backgroundColor,
-          textColor: textColor,
-          counter: counter,
-          onColorButtonPressed: _changeColor,
-          onCounterTextTap: () =>
-              _changeColorText(_colorLogic.generateRandomColor()),
+    return ChangeNotifierProvider<ColorLogic>(
+      create: (context) => serviceLocator.get<ColorLogic>(param1: email),
+      child: Consumer<ColorLogic>(
+        builder: (context,colorLogic, child) => GestureDetector(
+          onTap: colorLogic.changeAllColors,
+          child: Scaffold(
+            appBar: _ColorAppBar(
+              appBarColor: colorLogic.appBarColor,
+              email: email,
+              onTap: () {
```

Comment:
move resetCounter in handleAppBarTap, so in UI it would look like onTap: colorLogic.handleAppBarTap